Ladybird/Qt/Tab.cpp

```diff
@@ -893,15 +894,28 @@ void Tab::resizeEvent(QResizeEvent* event)
 
 void Tab::update_hover_label()
 {
+    m_hover_label->setText(QFontMetrics(m_hover_label->font()).elidedText(m_hover_label->text(), Qt::ElideRight, width() / 2));
     m_hover_label->resize(QFontMetrics(m_hover_label->font()).boundingRect(m_hover_label->text()).adjusted(-4, -2, 4, 2).size());
-    auto hover_label_height = height() - m_hover_label->height() - 8;
+
+    auto hover_label_height = height() - m_hover_label->height();
     if (m_find_in_page->isVisible())
-        hover_label_height -= m_find_in_page->height() - 4;
+        hover_label_height -= m_find_in_page->height() - 8;
+
+    QPoint mousePosition = mapFromGlobal(QCursor::pos());
+    if (mousePosition.y() >= hover_label_height && mousePosition.x() <= width() / 2)
+        m_hover_label->move(width() / 2 + (width() / 2 - m_hover_label->width()), hover_label_height);
+    else
+        m_hover_label->move(0, hover_label_height);
 
-    m_hover_label->move(6, hover_label_height);
     m_hover_label->raise();
 }
 
+void Tab::mouseMoveEvent(QMouseEvent* event)
+{
+    update_hover_label();
```

Comment:
It feels a bit excessive to call this method on every mouse move event. I think we should be able to get away with calling this only when the mouse is over the label. To do this you could make `m_hover_label` a subclass of `QLabel` and override `enterEvent()` like this:

```c++
class HyperlinkLabel final : public QLabel {
    Q_OBJECT
public:
    explicit HyperlinkLabel(QWidget *parent = nullptr)
        : QLabel(parent)
    {
        setMouseTracking(true);
    }

    virtual void enterEvent(QEnterEvent* event) override
    {
        emit mouse_entered(event);
    }

signals:
    void mouse_entered(QEnterEvent*);
    
};
```

Then, in `Tab.cpp` you can call `update_label()` when the `mouse_entered` signal is fired like this:
```c++
    QObject::connect(m_hover_label, &HyperlinkLabel::mouse_entered, [this] {
        update_hover_label();
    });
```

Reply:
Yes, I also thought that it might even hinder the performance, but couldn't come up with a way. Thank you for the suggestion and indepth review